experimental/traffic-portal/src/app/api/type.service.ts

```diff
@@ -79,6 +79,37 @@ export class TypeService extends APIService {
 		return this.getTypesInTable("server");
 	}
 
+	/**
+	 * Deletes an existing type.
+	 *
+	 * @param id ID of the type to delete.
+	 * @returns The deleted type.
+	 */
+	public async deleteType(id: number): Promise<TypeFromResponse> {
```

Comment:
Ideally anywhere you can pass the ID, you should also be able to pass the response form of the object (i.e. `TypeFromResponse` in this case).